contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+    /// @param _recipientId Id of the recipient
+    function getRecipientStatus(address _recipientId) external view override returns (RecipientStatus) {
+        InternalRecipientStatus internalStatus = _getRecipient(_recipientId).recipientStatus;
+        if (internalStatus == InternalRecipientStatus.Appealed) {
+            return RecipientStatus.Pending;
+        } else {
+            return RecipientStatus(uint8(internalStatus));
+        }
+    }
+
+    /// @notice Returns the payout summary for the accepted recipient
+    function getPayouts(address[] memory __recipientIds, bytes memory, address)
+        external
+        view
+        returns (PayoutSummary[] memory payouts)
+    {
+        uint256 recipientLength = __recipientIds.length;
+
+        payouts = new PayoutSummary[](recipientLength);
+
+        for (uint256 i = 0; i < recipientLength;) {
+            payouts[i] = payoutSummaries[__recipientIds[i]];
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Checks if address is elgible allocator
+    function isValidAllocator(address) external pure returns (bool) {
+        return true;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Review recipient application
+    /// @param __recipientIds Ids of the recipients
+    /// @param _recipientStatuses Statuses of the recipients
+    function reviewRecipients(address[] calldata __recipientIds, InternalRecipientStatus[] calldata _recipientStatuses)
+        external
+        onlyPoolManager(msg.sender)
+        onlyActiveRegistration
+    {
+        uint256 recipientLength = __recipientIds.length;
+        if (recipientLength != _recipientStatuses.length) {
+            revert INVALID();
+        }
+
+        for (uint256 i = 0; i < recipientLength;) {
+            if (
+                _recipientStatuses[i] == InternalRecipientStatus.None
+                    || _recipientStatuses[i] == InternalRecipientStatus.Appealed
+            ) {
+                revert INVALID();
+            }
+
+            Recipient storage recipient = _recipients[__recipientIds[i]];
+            recipient.recipientStatus = _recipientStatuses[i];
+
+            emit Reviewed(__recipientIds[i], recipient.recipientStatus, msg.sender);
+
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Set payout for the recipients
+    /// @param __recipientIds Ids of the recipients
+    /// @param _amounts Amounts to be paid out
+    function setPayout(address[] memory __recipientIds, uint256[] memory _amounts)
+        external
+        onlyPoolManager(msg.sender)
+        onlyAfterAllocation
+    {
+        uint256 recipientLength = __recipientIds.length;
+        if (recipientLength != _amounts.length) {
+            revert INVALID();
+        }
+
+        for (uint256 i = 0; i < recipientLength;) {
```

Comment:
check:
- if amount is already set and only set if amount has changed. 
- if we haven't reached the max pool amount
- recipient wasn't paid yet